e2e_test/batch/transaction/now.slt

```diff
@@ -0,0 +1,52 @@
+statement ok
+create table t (ts timestamp);
+
+statement ok
+create view v as select * from t where ts at time zone 'utc' >= now();
+
+statement ok
+create materialized view mv as select * from v;
+
+statement ok
+insert into t select * from generate_series(
+    now() at time zone 'utc' - interval '10' second,
+    now() at time zone 'utc' + interval '20' second,
+    interval '1' second / 20
+);
+
+statement ok
+flush;
+
+statement ok
+start transaction read only;
+
+# the view should not be empty, so that the following check is meaningful
+query T
+select count(*) > 0 from mv;
+----
+t
+
+# the result from batch query and materialized view should be the same
+query T
+select * from v
+except
+select * from mv;
```

Comment:
This test fails as we always yield the timestamp of the **next** epoch in `StreamNow` executor, before yielding the barrier message.

https://github.com/risingwavelabs/risingwave/blob/212cad4866ced33e1b1c960bfc1d8362e48d8239/src/stream/src/executor/now.rs#L98
https://github.com/risingwavelabs/risingwave/blob/212cad4866ced33e1b1c960bfc1d8362e48d8239/src/stream/src/executor/now.rs#L147